### PR TITLE
[SQL] Support MySQL IF&IFNULL Function

### LINKED ISSUE #454: [SQL] Support MySQL function
**What would you like to be added**:

**Why is this needed**:

**math function**
- [x] ABS
- [x] CEIL
- [x] FLOOR
- [x] RAND
- [x] SIGN
- [x] PI
- [x] TRUNCATE
- [x] ROUND
- [x] POWER
- [x] SQRT
- [x] EXP
- [x] MOD

**cast function**
- [x] CAST_UNSIGNED
- [x] #465 
- [x] CAST_CHARSET
- [x] CAST_NCHAR
- [x] CAST_CHAR
- [x] #569
- [x] #568
- [x] #570
- [x] #571

**IF function**
- [x] #493 

**crypto function**
- [x] MD5
- [x] #500
- [x] SHA1

**String function**
- [x] #501
- [x] #502
- [x] #503
- [x] #505 
- [x] #513
- [x] #517
- [x] #518
- [x] #519
- [x] #520
- [x] #537
- [x] #538
- [x] #539
- [x] #542
- [x] #543
- [x] #552
- [x] #553
- [x] #563
- [x] #564

### LINKED ISSUE #493: [SQL] Support MySQL `IF` & `IFNULL` function.
**What would you like to be added**:

**Why is this needed**:

ref #454 

official document:
https://dev.mysql.com/doc/refman/5.6/en/flow-control-functions.html#function_if
https://dev.mysql.com/doc/refman/5.6/en/flow-control-functions.html#function_ifnull

code:
https://github.com/arana-db/arana/tree/master/pkg/runtime/function/scripts/if.js

reference code:

https://github.com/arana-db/arana/tree/master/pkg/runtime/function2/abs.go

### PR DESCRIPTION
**Pull request by gongna-au**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #493 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/runtime/function/if.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	val3, err := inputs[2].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	isTrue, err := val1.Int64()
```
Comment (jjeffcaii) [inline pkg/runtime/function/if.go:58]: Should check `val1 != nil` first, then check `v, _ := val1.Int64(); v != 0`.

File: pkg/runtime/function/if_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+package function
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/arana-db/arana/pkg/proto"
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/assert"
```
Comment (jjeffcaii) [inline pkg/runtime/function/if_test.go:27]: reformat imports.

**Review by jjeffcaii — COMMENTED** [review]
pls fix

**Review by dongzl — APPROVED** [review]

File: pkg/runtime/function/if.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+type ifFunc struct{}
+
+func (i ifFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	val1, err := inputs[0].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	if val1 == nil {
```
Comment (jjeffcaii) [inline pkg/runtime/function/if.go:48]: image
I tested the case of `NULL` as first input argument, it seems returns the second branch, pls check it.

**Review by jjeffcaii — COMMENTED** [review]
pls fix

**Review by jjeffcaii — APPROVED** [review]

**Pull request merged**